### PR TITLE
cloud config for aws

### PR DESCRIPTION
**Pull request by gzt5142**
First draft of the AWS config notes related to kubernetes clusters on esip/qhub. 

@rsignell-usgs -- please review and correct/expand as you thing appropriate.   We should also talk about whether most of this can be buried into a helper function and what that might look like.  

@amsnyder  -- I created a new file in a location where I thought it might fit.  If it's in the wrong spot in the directory tree, we should be able to move it easily with a commit to this branch before merging.

### CODE REVIEW

**Comment by gzt5142** [comment]
Closes #155

File: essential_reading/ParallelProcessing/Clusters_AWS_Credentials.md [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+
+It is critical to understand that those workers get a copy of the program
+*only* and run it in a lightweight Docker container holding the conda
+environment.  Cluster workers under this scheme do **NOT** get a copy of the
+operating system, and they do not have access to a file system. Individual
+cluster workers cannot read or write to (for example) `/home/username/file.txt`
+because `/home` is not availabie; no filesystems are available to node
+workers on a cloud cluster.
+
+What does this mean for AWS credentials?  Typicall, those credentials are
```
Comment (rsignell-usgs) [inline essential_reading/ParallelProcessing/Clusters_AWS_Credentials.md:52]: "Typically"

**Review by rsignell-usgs — COMMENTED** [review]
Gene, I think this looks great!

**Comment by gzt5142** [comment]
> We should also talk about whether most of this can be buried into a helper function and what that might look like.


@rsignell-usgs I'd like to get your opinion on this...  Do we want to continue to offer 'helper' scripts to do some of this tedious work, or do we want to keep the credential management as a code cell in every notebook?  

Check out the example at: <https://jupyter.qhub.esipfed.org/hub/user-redirect/lab/tree/shared/users/gzt5142/ClustDemo_v2.ipynb>

This demonstrates what I think is a reasonable way to hide some of the details of AWS credentials and cluster config. The reason I like this is that it has more complicated code than we probably want to put in front of users... notably, this latest revision will check to see if AWS secrets are already set before it tries to read the credentials file.  That is how I've chosen to solve the problem of running a notebook as a kbatch submission.  With kbatch, if the AWS secrets are already in the environment, this helper will respect them and not try to overwrite by consulting the credentials.

**Comment by rsignell-usgs** [comment]
@gzt5142 I love the cleanness of this, but I think the problem is that many notebooks want a different set of options for the cluster.  Sometimes you want autoscaling with min4, max10, sometimes you want 30, sometimes you want small workers, sometimes large, sometimes you want to wait for all the workers, sometimes you don't.   This would require a bunch of different .py scripts, right, one for each option?   And then it wouldn't be clear to the notebook user what was being used, right.  

I guess I don't see how to get around having a function like we currently have.

**Comment by gzt5142** [comment]
@rsignell-usgs 

The main question in my mind is whether or not we want a 'helper' system for **AWS credential configuration**:  
> [Do we want to continue to offer 'helper' scripts to do some of this tedious work, or do we want to keep the credential management as a code cell in every notebook?](https://github.com/hytest-org/hytest/pull/156#issuecomment-1362090596)

The cluster parameter question is a separate (but related) issue. And... we have a few options to parameterize cluster spin-up without having to create separate `.py` files for each set of options.  I agree with you that most of these options will be invoked with mechanics similar to the imported `ebd` module currently used in a lot of notebooks. 

So I'm still unclear as to what you are recommending for the standard for HyTEST tutorials as to how I should model the configuration of AWS credentials. Should I put a block of code in an early cell (to parse the config file), import a custom module, or use `ebd` ? Or can I make it a case-by-case decision?  Case-by-case is an attractive option, but has negative implications for a tutorial (where consistency has value).

**Comment by rsignell-usgs** [comment]
@gzt5142 it seems a helper .py file would be useful to set credentials.   There are really no complicated options to set like on the cluster config.   Do you agree?

**Review by alaws-USGS — APPROVED** [review]
No security vulnerabilities seen, clean code.

**Pull request merged**